scripts/spack/configs/blueos_3_ppc64le_ib_p9/packages.yaml

```diff
@@ -72,37 +72,37 @@ packages:
   # System level packages to not build
   autoconf:
     externals:
-      - spec: autoconf
+      - spec: autoconf@2.69
         prefix: /usr
     buildable: false 
   automake:
     externals:
-      - spec: automake
+      - spec: automake@1.13.4
         prefix: /usr
     buildable: false
   bzip2:
     externals:
-      - spec: bzip2
+      - spec: bzip2@1.0.6
         prefix: /usr
     buildable: false
   gettext:
     externals:
-      - spec: gettext
+      - spec: gettext@0.19.8.1
         prefix: /usr
     buildable: false
   libtool:
     externals:
-      - spec: libtool
+      - spec: libtool@2.4.2
         prefix: /usr
     buildable: false
   m4:
     externals:
-      - spec: m4
+      - spec: m4@1.4.16
```

Comment:
@becker33 is there any way around this? Bumping these versions is going to be tedious.  Not to mention this probably isn't accurate across multiple machines...

Reply:
@white238 I think you should be able to remove that entirely and run `spack external find --scope site --not-buildable m4` to generate the config for m4. That won't be true for every package you have in these files, but it is true for many of them (basically all of the build tools, for example).

Reply:
You might not even need the scope argument, I'm not remembering exactly how you munge the configs inside uberenv, so I was trying to be careful with it.

Reply:
The problem is there are occasionally packages that actually care what version it is. I don't think I've seen that for m4, but I have for automake, autoconf, and bison.

What the old concretizer did was just assume it was the newest version Spack was aware of if it didn't have a version -- if that was a problem, you just got heinous errors, but that was relatively uncommon. I can look into how hard that would be to implement in the new concretizer.

Actually, maybe a better solution would be to allow externals not to have a version -- then you get an error if any package cares what version it was, but otherwise it can be fine without. That would probably be pretty complicated for us to implement, though, so that's a downside.